### ISSUE TITLE
Occurrence numbers should give an error when there are not enough matches

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#13566]
Shouldn't they?

```
Require Setoid.
Goal 1 = 0 -> 0 = 1.
intro H.
rewrite H at 1 2 3.
```

**Comment by herbelin** [comment]
Fixed in #13056.

In passing, I'm dreaming of a nice unifying model for the different variants of `rewrite` which would both be clearer for users and carrying less technical debt (cc @mattam82).

**Comment by jfehrle** [comment]
Great!  I think you meant to refer to #13566 in your comment, which I edited.

> In passing, I'm dreaming of a nice unifying model for the different variants of `rewrite`

Sounds good.
